activesupport/lib/active_support/core_ext/string/inflections.rb

```diff
@@ -164,8 +164,29 @@ def deconstantize
   #
   #   <%= link_to(@person.name, person_path) %>
   #   # => <a href="/person/1-donald-e-knuth">Donald E. Knuth</a>
-  def parameterize(sep = '-'.freeze)
-    ActiveSupport::Inflector.parameterize(self, sep)
+  def parameterize(sep = '-'.freeze, cas = :downcase)
+    ActiveSupport::Inflector.parameterize(self, sep, cas)
+  end
+
+  # Forces string to downcase after replacing all special characters
+  #
+  # "Donald E. Knuth".parameterize_downcase => donald-e-knuth
+  def parameterize_downcase(sep = '-'.freeze, cas = :downcase)
+    ActiveSupport::Inflector.parameterize(self, sep, cas)
+  end
+
+  # Forces string to upcase after replacing all special characters
+  #
+  # "Donald E. Knuth".parameterize_upcase => DONALD-E-KNUTH
+  def parameterize_upcase(sep = '-'.freeze, cas = :upcase)
+    ActiveSupport::Inflector.parameterize(self, sep, cas)
+  end
+
+  # Retains character case after replacing all special characters
+  #
+  # "Donald E. Knuth".parameterize_withcase => Donald-E-Knuth
+  def parameterize_withcase(sep = '-'.freeze, cas = :withcase)
+    ActiveSupport::Inflector.parameterize(self, sep, cas)
   end
```

Comment:
Writing your same code in this style, I think would make it more nicer.

Although it doesn't matter its really a very small change

```
def parameterize_withcase(sep = '-'.freeze, cas = :downcase)
  ActiveSupport::Inflector.parameterize(self, sep, cas)
end

def parameterize_downcase(sep = '-'.freeze)
  parameterize_withcase(sep,:downcase)
end

def parameterize_upcase(sep = '-'.freeze)
  parameterize_withcase(sep,:upcase)
end
```